kv/key.go

```diff
@@ -434,16 +465,32 @@ func (m *HandleMap) Set(h Handle, val interface{}) {
 
 // Delete deletes a entry from the map.
 func (m *HandleMap) Delete(h Handle) {
+	ints, strs := m.ints, m.strs
+	if ph, ok := h.(PartitionHandle); ok {
+		idx := ph.PartitionID
+		if (h.IsInt() && m.partitionInts[idx] == nil) ||
+			(!h.IsInt() && m.partitionStrs[idx] == nil) {
+			return
+		}
+		ints, strs = m.partitionInts[idx], m.partitionStrs[idx]
+	}
 	if h.IsInt() {
-		delete(m.ints, h.IntValue())
+		delete(ints, h.IntValue())
 	} else {
-		delete(m.strs, string(h.Encoded()))
+		delete(strs, string(h.Encoded()))
 	}
 }
 
 // Len returns the length of the map.
 func (m *HandleMap) Len() int {
```

Comment:
Can `ints` and `partitionInts` merge into one? Otherwise `Len()` seems strage, because `ints` and `partitionInts` are exclusive.

Reply:
I think we could use `partitionInts[0]` to instead of `ints`, because `partitionID` shouldn't equal to 0. But i don't think that's good
1. an intHandle will do one more map lookup
2. implicitly assumes that partitionID will not be equal to 0, a little bit hack here

And I don't get why `Len()` function seems strange here. Because we could store different handle types in one `HandleMap`. Just like `TestHandleMapWithPartialHandle` test.